test/inductor/test_dropout_align_random_eager.py

```diff
@@ -0,0 +1,252 @@
+# Torch-Inductor knobs
+torch._inductor.config.triton.unique_kernel_names = True
+torch._inductor.config.coordinate_descent_tuning = True
+torch._inductor.config.freezing = True
+torch._inductor.config.align_random_eager = True
+# torch._inductor.config.fallback_random = True
+# torch._inductor.config.max_autotune_pointwise = True
+
+
+# ───────────────────────────────────────────────────────────────
+# Model under test
+# ───────────────────────────────────────────────────────────────
+class LinearBlock(torch.nn.Module):
+    def __init__(self, hidden_dim: int, ffn_dim: int, dropout: float = DROPOUT_P):
+        super().__init__()
+        self.net = torch.nn.Sequential(
+            torch.nn.Linear(hidden_dim, ffn_dim),
+            torch.nn.Dropout(dropout),
+            torch.nn.ReLU(inplace=False),
+            torch.nn.Linear(ffn_dim, hidden_dim),
+        )
+
+    def forward(self, x: torch.Tensor):
+        return self.net(x)
+
+
+def build_models(dropout: float):
+    eager = LinearBlock(HIDDEN_DIM, FFN_DIM, dropout)
+    compiled = LinearBlock(HIDDEN_DIM, FFN_DIM, dropout)
+    compiled.load_state_dict(eager.state_dict())
+    compiled = torch.compile(compiled)
+    return eager, compiled
+
+
+# ───────────────────────────────────────────────────────────────
+# Pytest fixtures
+# ───────────────────────────────────────────────────────────────
+@pytest.fixture(autouse=True)
+def _set_seed():
+    torch.manual_seed(BASE_SEED)
+    if torch.cuda.is_available():
+        torch.cuda.manual_seed(BASE_SEED)
+    yield
+
+
+@pytest.fixture(params=["cpu"] + (["cuda"] if torch.cuda.is_available() else []))
+def device(request):
+    dev = torch.device(request.param)
+    if dev.type == "cuda" and not torch.cuda.is_available():
+        pytest.skip("CUDA not available")
+    return dev
+
+
+@pytest.fixture
+def input_tensor(device):
+    return torch.randn(BATCH, SEQ_LEN, HIDDEN_DIM, device=device)
+
+
+# ───────────────────────────────────────────────────────────────
+# Helpers
+# ───────────────────────────────────────────────────────────────
+def _sync(x: torch.Tensor):
+    if x.is_cuda:
+        torch.cuda.synchronize()
+
+
+def _timed_run(model, x, backward: bool = False):
+    _sync(x)
+    t0 = time.time()
+    y = model(x)
+    if backward:
+        (y.square().mean()).backward()
+    _sync(x)
+    return (time.time() - t0) * 1e3, y
+
+
+def _cuda_rng_u64_seed_off():
+    """Return (seed, offset) extracted from torch.cuda.get_rng_state()."""
+    st = torch.cuda.get_rng_state()
+    seed = struct.unpack("<Q", st[0:8].cpu().numpy().tobytes())[0]
+    off = struct.unpack("<Q", st[8:24].cpu().numpy().tobytes())[0]
+    return seed, off
+
+
+# ───────────────────────────────────────────────────────────────
+# Tests: correctness (forward/backward)
+# ───────────────────────────────────────────────────────────────
+@pytest.mark.parametrize("training", [False, True])
+def test_linear_block_compile_parity_forward(device, input_tensor, training):
```

Comment:
Can you please copy the test inductor idioms for test classes - see:

https://github.com/pytorch/pytorch/blob/aa4a8c9b9228a8999cac1fba37cd04e086ecfb9b/test/inductor/test_control_deps.py#L2-L16